BUILDING.md

```diff
@@ -16,7 +16,7 @@ following packages are installed:
 
 Binary builds additionally require the following packages to be
 available to the CMake build system.
-- Python 3.9 (development install)
+- Python 3.7 (development install)
```

Comment:
[plan] In order to test this PR I wanted to run 
```
conan install -if .conan --require-override cpython/3.7.12 resources/build
````
i.e., making use of `--require-override` to force use of 3.7. 

However, our conanfile has a mistake (my bad), similar to that alluded to in https://github.com/OpenAssetIO/OpenAssetIO/issues/654#issuecomment-1290756213, but for other reasons.  

Since `cpython` is a `tool_requires` rather than normal `requires`, the `--require-override` has no effect. [The docs](https://docs.conan.io/en/latest/reference/commands/consumer/install.html#require-override) say this explicitly

> This feature affects only to regular requires, not to tool_requires or python_requires, as those don’t have such an overriding mechanism, and they are private to their consumer, not propagating downstream nor upstream.

We need to split our `build_requirements` method in `conanfile.py` into separate `requirements` and `build_requirements`, where the `requirements` method contains `requires()` calls for `cpython` and `pybind11`, and the `build_requirements` method contains the `tool_requires()` calls for the unit testing libraries.

Reply:
https://github.com/OpenAssetIO/OpenAssetIO/issues/699